Can't used mirrored, translatable type as translated HashMap value

### Describe the bug

In integrating FRB 2.0 (amazing to see this shaping up!), a mirrored, translatable type can't be used as a mirrored, translatable HashMap value due to a missing IntoDart trait bound on the outside crate's type.

### Steps to reproduce

Other crate, imported:

```
#[derive(Clone)]
pub struct TestInner {}
#[derive(Clone)]
pub struct TestOuter {
    pub test_inner: HashMap<String, TestInner>,
}
```

FRB Rust crate:
```
#[frb(mirror(TestInner))]
pub struct _TestInner {

}

#[frb(mirror(TestOuter))]
pub struct _TestOuter {
    pub test_inner: HashMap<String, TestInner>,
}
```

Error message from `flutter_rust_bridge_codegen generate && flutter_rust_bridge_codegen build-web`

```
error[E0599]: the method `into_into_dart` exists for struct `HashMap<String, TestInner>`, but its trait bounds were not satisfied
    --> front/getplans_io/rust/src/frb_generated.rs:1234:28
     |
1234 |         [self.0.test_inner.into_into_dart().into_dart()].into_dart()
     |                            ^^^^^^^^^^^^^^ method cannot be called on `HashMap<String, TestInner>` due to unsatisfied trait bounds
     |
    ::: /Users/aran/Projects/plans/code/back/editor/commands/src/lib.rs:177:1
     |
177  | pub struct TestInner {}
     | -------------------- doesn't satisfy `TestInner: IntoDart`
     |
    ::: /Users/aran/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/src/rust/library/std/src/collections/hash/map.rs:214:1
     |
214  | pub struct HashMap<K, V, S = RandomState> {
     | ----------------------------------------- doesn't satisfy `_: IntoIntoDart<HashMap<String, TestInner>>`
     |
     = note: the following trait bounds were not satisfied:
             `TestInner: IntoDart`
             which is required by `HashMap<std::string::String, TestInner>: IntoIntoDart<HashMap<std::string::String, TestInner>>`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `rust_lib` (lib) due to previous error
```



### Logs

```shell
N/A
```


### Expected behavior

_No response_

### Generated binding code

_No response_

### OS

macos

### Version of `flutter_rust_bridge_codegen`

2.0.0-dev.16

### Flutter info

```shell
N/A
```


### Version of `clang++`

Apple clang version 15.0.0 (clang-1500.1.0.2.5)

### Additional context

_No response_

--- a/frb_example/pure_dart/rust/example_external_lib/src/lib.rs
+++ b/frb_example/pure_dart/rust/example_external_lib/src/lib.rs
@@ -1,6 +1,8 @@
 //! Examples of types you'd want to mirror
 //!
 
+use std::collections::HashMap;
+
 #[derive(Debug, Clone)]
 pub struct RawStringMirrored {
     pub r#value: String,
@@ -58,6 +60,14 @@ pub struct Sequences(pub Vec<i32>);
 
 #[derive(Debug, Clone)]
 pub struct ApplicationEnvVar(pub String, pub bool);
+
+#[derive(Debug, Clone)]
+pub struct HashMapValue {}
+
+#[derive(Debug, Clone)]
+pub struct StructWithHashMap {
+    pub map: HashMap<String, HashMapValue>,
+}
 
 impl ApplicationSettings {
     pub fn new(

--- a/frb_example/pure_dart/rust/src/api/mirror.rs
+++ b/frb_example/pure_dart/rust/src/api/mirror.rs
@@ -10,7 +10,9 @@ pub use frb_example_pure_dart_exapmle_external_lib::{
     ApplicationEnv, ApplicationEnvVar, ApplicationMessage, ApplicationMode, ApplicationSettings,
     ListOfNestedRawStringMirrored, NestedRawStringMirrored, Numbers, RawStringEnumMirrored,
     RawStringMirrored, Sequences,
+    HashMapValue, StructWithHashMap,
 };
+use std::collections::HashMap;
 
 // To mirror an external struct, you need to define a placeholder type with the same definition
 #[frb(mirror(ApplicationSettings))]
@@ -34,6 +36,14 @@ pub struct _ApplicationEnvVarTwinNormal(pub String, pub bool);
 #[frb(mirror(ApplicationEnv))]
 pub struct _ApplicationEnvTwinNormal {
     pub vars: Vec<ApplicationEnvVar>,
+}
+
+#[frb(mirror(HashMapValue))]
+pub struct _HashMapValue {}
+
+#[frb(mirror(StructWithHashMap))]
+pub struct _StructWithHashMap {
+    pub map: HashMap<String, HashMapValue>,
 }
 
 // This function can directly return an object of the external type ApplicationSettings because it has a mirror
@@ -233,5 +243,13 @@ pub fn test_contains_mirrored_sub_struct_twin_normal() -> ContainsMirroredSubStr
         test2: AnotherTwinNormal {
             a: "test".to_owned(),
         },
+    }
+}
+
+pub fn test_hashmap_with_mirrored_value() -> StructWithHashMap {
+    StructWithHashMap {
+        map: {
+            HashMap::new()
+        }
     }
 }

--- a/frb_rust/src/misc/into_into_dart.rs
+++ b/frb_rust/src/misc/into_into_dart.rs
@@ -75,14 +75,19 @@ where
     }
 }
 
-impl<K, V> IntoIntoDart<HashMap<K, V>> for HashMap<K, V>
+impl<KT, KD, VT, VD> IntoIntoDart<HashMap<KD, VD>> for HashMap<KT, VT>
 where
-    K: IntoDart,
-    V: IntoDart,
+    KT: IntoIntoDart<KD>,
+    VT: IntoIntoDart<VD>,
+    HashMap<KD, VD>: IntoDart,
+    KD: IntoDart + std::cmp::Eq + std::hash::Hash,
+    VD: IntoDart,
 {
     #[inline(always)]
-    fn into_into_dart(self) -> Self {
-        self
+    fn into_into_dart(self) -> HashMap<KD, VD> {
+        self.into_iter()
+            .map(|(k, v)| (k.into_into_dart(), v.into_into_dart()))
+            .collect()
     }
 }
 

